### DISCUSSION TITLE
Unique constraint field error is gone after v0.14

### DISCUSSION THREAD

**Discussion by fmaclen** [pocketbase/pocketbase#2287] [Q&A] [answered]
I'm in the middle of the upgrade process from `v0.13` to `v0.14` but I'm running into an issue regarding the response I get when a column is `unique`.

For duplicate emails on the users table I get the following error which works as expected:
```json
{
  "url": "http://127.0.0.1:8091/api/collections/users/records",
  "status": 400,
  "data": {
    "code": 400,
    "message": "Failed to create record.",
    "data": {
      "email": {
        "code": "validation_invalid_email",
        "message": "The email is invalid or already in use."
      }
    }
  },
  "isAbort": false,
  "originalError": null,
  "name": "ClientResponseError 400"
}
```

But in my case I also have a column called `nickname` that was _marked as unique_ in `v0.13`.
This is what it looks like after it got migrated by `v0.14`:

![image](https://user-images.githubusercontent.com/1434675/231851969-407ed882-21c0-44bc-a6d3-ab6025d16a0b.png)

Now, if I submit a form with a duplicate nickname I get an error with an empty `data { }`:

```json
{
  "url": "http://127.0.0.1:8090/api/collections/users/records",
  "status": 400,
  "data": {
    "code": 400,
    "message": "Failed to create record.",
    "data": {}
  },
  "isAbort": false,
  "originalError": null,
  "name": "ClientResponseError 400"
}
```

My app was expecting a `data: { nickname: { code: "...", message: "..." } }` error so it can surface the message to the user on the UI.

I'm not sure if this is a bug on my end or if that behavior was dropped from PB in the last update.

**Comment by ganigeorgiev** [comment]
Hm, this looks like a bug with the errors propagation since it works when the request is admin token authenticated. I'll investigate it in more details tomorrow and will try to submit a fix in v0.14.5.

**Reply by ganigeorgiev** [reply, marked as answer]
It should be fixed in [v0.14.5 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.14.5).
_Note that if the unique index is constructed from a custom expression (eg. concatenating 2 columns in 1), you still will get a generic error._

**Reply by fmaclen** [reply]
Perfect, `v0.14.5` worked!

> you still will get a generic error

That's alright, in my case I was overriding the default message with my own custom one anyways 👍 

There is however one more issue I wanted to ask you about. When I ran `v0.14` it correctly updated the schema as shown in the screenshot of my first post, however, that process did not create a new migration in `/pb_migrations`.

Which means that if I want to create a new DB from scratch with the same schema I don't get the unique constraints.

For context, I'm working on a SvelteKit app and for testing I'm [creating a copy of `/pb_migrations`](https://github.com/fmaclen/promptspree/blob/main/scripts/pocketbaseSetupTest.js#L24-L32) on the `/tests` folder, so every time I run Playwright I get a new blank DB, but with the correct schema.

Does that make sense?

**Reply by ganigeorgiev** [reply]
> Which means that if I want to create a new DB from scratch with the same schema I don't get the unique constraints.

I'm not sure that I understand what do you mean.

When migrating from v0.13 to v0.14 there is system migration (https://github.com/pocketbase/pocketbase/blob/master/migrations/1679943781_add_indexes_column.go) that takes care for converting the unique checks to unique indexes.

This migration is embedded in the  executable. Every time when you start the application with a new db, we execute both any unapplied system and user migrations.

If you want to have the current indexes exported as migration in your `/pb_migrations` directory, you can run `./pocketbase migrate collections`. This will create a new migration file with full snapshot of your current collections configuration.

**Reply by fmaclen** [reply]
> you can run ./pocketbase migrate collections

Gotcha, this solves my issue 👍 

What I meant earlier is that whenever you make a change using the UI the app generates a new migration file, so I was expecting that to also be the case with these new unique constraints/indexes.
